timeutil/schedule.go

```diff
@@ -150,34 +154,159 @@ func findNthWeekDay(t time.Time, weekday time.Weekday, nthInMonth uint) time.Tim
 	return t
 }
 
+// findLastWeekDay finds the last occurrence of a given weekday in the month of t
+func findLastWeekDay(t time.Time, weekday time.Weekday) time.Time {
+	n := monthNext(t).Add(-24 * time.Hour)
+	for n.Weekday() != weekday {
+		n = n.Add(-24 * time.Hour)
+	}
+	return n
+}
+
+// matchingWeekdaysInMonth returns the number of occurrences of the weekday of t since
+// the start of the month until t event
+func matchingWeekdaysInMonth(t time.Time) int {
+	month := t.Month()
+	nth := 0
+	for n := t; n.Month() == month; n = n.Add(-7 * 24 * time.Hour) {
+		nth++
+	}
+	return nth
+}
+
 // Match checks if t is within the day span represented by ws.
 func (ws WeekSpan) Match(t time.Time) bool {
 	start, end := ws.Start, ws.End
 	wdStart, wdEnd := start.Weekday, end.Weekday
 
-	if start.Pos != EveryWeek {
-		if start.Pos == LastWeek {
-			// last week of the month
-			if !isLastWeekdayInMonth(t) {
+	weekdayMatch := func(t time.Time) bool {
+		if wdStart <= wdEnd {
+			// single day (mon) or start < end (eg. mon-fri)
+			return t.Weekday() >= wdStart && t.Weekday() <= wdEnd
+		}
+		// wraps around the week end, eg. fri-mon
+		return t.Weekday() >= wdStart || t.Weekday() <= wdEnd
+	}
+
+	if start.Pos == EveryWeek && end.Pos == EveryWeek {
+		// generic weekday match, eg. mon-fri
+		return weekdayMatch(t)
+	}
+
+	// things that use a numbered weekday
+
+	// fun cases, eg (consider the calendar below):
+	//
+	// - mon1-fri, week span, start anchored at 1st Monday 06.08, matches:
+	// 06.08-10.08
+	// - mon-fri2, week span, end anchored at 2nd Friday 10.08, matches:
+	// 06.08-10.08
+	// - fri1-mon, week span, start anchored at 1st Friday 3.08, matches
+	// 03.08-06.08
+	// - mon-fri1, week span, end anchored at 1st Friday 3.08, matches
+	// 30.07-03.08, (crossing the month boundary)
+	// - fri4-thu, week span, end anchored at 4th Friday 27.07, matches
+	// 27.07-02.08, (crossing the month boundary)
```

Comment:
note this also matches `24.08-30.08`, maybe add this in the comment explaining what matches

Reply:
Added